ql/models/marketmodels/driftcomputation/smmdriftcalculator.cpp

```diff
@@ -83,14 +83,19 @@ namespace QuantLib {
         // assuming terminal bond measure
         // eq 5.4-5.7
         const std::vector<Time>& taus=cs.rateTaus();
+        std::vector<Real> annuities(numberOfRates_);
+
+        for (Size j=0; j<numberOfRates_; ++j) {
+            annuities[j] = cs.coterminalSwapAnnuity(numberOfRates_, j);
+        }
         for (Size k=0; k<numberOfFactors_; ++k) {
                 // taken care in the constructor
                 // wkpj1_[k][numberOfRates_-1]= 0.0;
                 // wkaj_[k][numberOfRates_-1] = 0.0;
             for (Integer j=numberOfRates_-2; j>=static_cast<Integer>(alive_)-1; --j) {
                  // < W(k) | P(j+1)/P(n) > =
                  // = SR(j+1) a(j+1,k) A(j+1) / P(n) + SR(j+1) < W(k) | A(j+1)/P(n) >
-                Real annuity = cs.coterminalSwapAnnuity(numberOfRates_,j+1);
+                Real annuity = annuities[j+1];
```

Comment:
can the same vector `annuities` be used also in the other loop ~20 lines below?

Reply:
Yes! I'll make that change too, thanks for pointing it out. In fact the quantitative results stated above were including this change, but I forgot to add it when making the final PR.

Reply:
Great, thanks.